geniza/corpus/management/commands/tei_to_annotation.py

```diff
@@ -0,0 +1,347 @@
+            # found the document
+            if self.verbosity >= self.v_normal:
+                self.stdout.write(doc)
+
+            # get the footnote for this file & doc
+            footnote = self.get_edition_footnote(doc, tei, xmlfile)
+            # if no footnote, skip for now
+            # TODO: generate footnote when needed! (but should we need to?)
+            if not footnote:
+                self.stdout.write(
+                    self.style.ERROR(
+                        "footnote not found for %s / %s; skipping" % (xmlfile, doc.pk)
+                    )
+                )
+                continue
+            footnote = self.migrate_footnote(footnote, doc)
+
+            # if there is a single primary language, use the iso code if it is set
+            lang_code = None
+            if doc.languages.count() == 1:
+                lang_code = doc.languages.first().iso_code
+
+            # get html blocks from the tei
+            html_blocks = tei.text_to_html(block_format=True)
+
+            # get canvas objects for the images in order; skip any non-document images
+            iiif_canvases = [img["canvas"] for img in doc.iiif_images(filter_side=True)]
+            # determine the number of canvases needed based on labels
+            # that indicate new pages
+            # check and count any after the first; always need at least 1 canvas
+            num_canvases = 1 + len(
+                [
+                    b["label"]
+                    for b in html_blocks[1:]
+                    if tei.label_indicates_new_page(b["label"])
+                ]
+            )
+            # in verbose mode report on available/needed canvases
+            if self.verbosity > self.v_normal:
+                self.stdout.write(
+                    "%d iiif canvases; need %d canvases for %d blocks"
+                    % (len(iiif_canvases), num_canvases, len(html_blocks))
+                )
+            # if we need more canvases than we have available,
+            # generate local canvas ids
+            if num_canvases > len(iiif_canvases):
+                # document manifest url is /documents/pgpid/iiif/manifest/
+                # create canvas uris parallel to that
+                canvas_base_uri = doc.manifest_uri.replace("manifest", "canvas")
+                for i in range(num_canvases - len(iiif_canvases)):
+                    canvas_uri = "%s%d/" % (canvas_base_uri, i + 1)
+                    iiif_canvases.append(canvas_uri)
+
+            # NOTE: pgpid 1390 folio example; each chunk should be half of the canvas
+            # (probably should be handled manually)
+            # if len(html_chunks) > len(iiif_canvases):
+            #     self.stdout.write(
+            #         "%s has more html chunks than canvases; skipping" % xmlfile
+            #     )
+            #     continue
+
+            # start attaching to first canvas; increment based on chunk label
+            canvas_index = 0
```

Comment:
I was wondering about this actually (and also for the generated canvases elsewhere), should this be 1-indexed? Seems that's the convention for most of the manifests, canvases, etc we're working with (and it's how we'll be ordering annotations now)

Reply:
No need I think, I just missed the `i + 1` when generating the canvases.